cider/exec/nextgen/jitlib/base/JITFunction.h

```diff
@@ -34,6 +34,7 @@ enum JITFunctionParamAttr : uint64_t {};
 struct JITFunctionParam {
   std::string name{""};
   JITTypeTag type;
+  JITTypeTag sub_type{JITTypeTag::INVALID};
```

Comment:
this will only be used when `type` is pointer, right? Please add some comments for that. BTW, maybe we need a class named `JITType` which have sub_type?

Reply:
> this will only be used when `type` is pointer, right? Please add some comments for that. BTW, maybe we need a class named `JITType` which have sub_type?

Yes, just for POINTER type. Seems class ```JITType``` is unnecessary because it is only used to hold type and sub_type.